### PR TITLE
Adding debug logger 

### PR DESCRIPTION
**Pull request by this**
For issue #175.

### CODE REVIEW

File: components/uuf-core/src/main/java/org/wso2/carbon/uuf/core/Fragment.java [hunk 1]
```diff
@@ -70,6 +71,8 @@ public String render(Model model, Lookup lookup, RequestLookup requestLookup, AP
             }
         }
 
+        // Debug logs for fragment rendering start.
+        DebugLogger.startFragment(this);
```
Comment (Kishanthan) [inline components/uuf-core/src/main/java/org/wso2/carbon/uuf/core/Fragment.java]: We should start and end "debug" using a try and finally block.

File: components/uuf-core/src/main/java/org/wso2/carbon/uuf/internal/debug/DebugLogger.java [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+    }
+
+    /**
+     * Logs the end of dispatching the specified HTTP request.
+     *
+     * @param request dispatched HTTP request
+     */
+    public static void endRequest(HttpRequest request) {
+        if (IS_DEBUGGING_ENABLED) {
+            logEvent(EventType.END_PAGE, request);
```
Comment (Kishanthan) [inline components/uuf-core/src/main/java/org/wso2/carbon/uuf/internal/debug/DebugLogger.java]: This should be "END_REQUEST" I believe ?

Comment (this) [inline components/uuf-core/src/main/java/org/wso2/carbon/uuf/internal/debug/DebugLogger.java]: fixed in commit b1b5aa69bb0c6b117aa48c7bd4fd6530a30944b3

File: components/uuf-core/src/main/java/org/wso2/carbon/uuf/core/Fragment.java (hunk 1 above)
Comment (this) [inline components/uuf-core/src/main/java/org/wso2/carbon/uuf/core/Fragment.java]: fixed in commit cd0c505f415bee2d307cb37034a97ff672eaccf5

**Comment by Kishanthan** [comment]
@msmshariq please merge this along with the #112

**Pull request merged**